zircon.core/src/main/kotlin/org/hexworks/zircon/internal/component/renderer/DefaultLogAreaRenderer.kt

```diff
@@ -0,0 +1,58 @@
+package org.hexworks.zircon.internal.component.renderer
+
+import org.hexworks.zircon.api.component.LogArea
+import org.hexworks.zircon.api.component.renderer.ComponentRenderContext
+import org.hexworks.zircon.api.component.renderer.ComponentRenderer
+import org.hexworks.zircon.api.data.Position
+import org.hexworks.zircon.api.data.Size
+import org.hexworks.zircon.api.graphics.SubTileGraphics
+import org.hexworks.zircon.internal.component.impl.log.LogTextElement
+
+class DefaultLogAreaRenderer : ComponentRenderer<LogArea>() {
+
+    override fun render(tileGraphics: SubTileGraphics, context: ComponentRenderContext<LogArea>) {
+        context.component.getLogElementBuffer().clearLogRenderPositions()
+
+        val style = context.componentStyle.currentStyle()
+        tileGraphics.applyStyle(style)
+        tileGraphics.clear()
+
+        val component = context.component
+        val logElements = component.getLogElementBuffer().getAllLogElements()
+        var currentLogElementY = 0
+        var currentScreenPosY = 0
+
+        var delayTimeInMs = if (isTypewriterEffectSupported(context))
+            context.component.delayInMsForTypewriterEffect.toLong()
+        else
+            0L
+
+        logElements.forEach { element ->
+            val result = element.render(tileGraphics,
+                    LogElementRenderContext(context, PositionRenderContext(currentScreenPosY, currentLogElementY), delayTimeInMs))
+            currentScreenPosY = result.currentScreenPosY
+            currentLogElementY = element.getPosition().y
+            delayTimeInMs = result.delayTimeInMs
+        }
+    }
+
+    private fun isTypewriterEffectSupported(context: ComponentRenderContext<LogArea>) =
```

Comment:
The typewriter effect can be simplified to setting `Delay` modifiers to all the `Tiles` based on their `x` positions at the instance they are added (in `DefaultLogArea`) so all other actors can be oblivious to their existence.